indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -8,951 +8,309 @@
-      !roomAttributes ||
-      !roomEquipment ||
-      !roomAvailability
-    );
-  }
-
-  async fetchAttributes() {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchAttributesURL());
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    this.setState({attributes: response.data});
-  }
-
-  async fetchRoomData(roomId) {
-    const reqs = [
-      this.fetchDetailedRoom(roomId),
-      this.fetchRoomAttributes(roomId),
-      this.fetchRoomAvailability(roomId),
-      this.fetchRoomEquipment(roomId),
-    ];
-    this.setState(Object.assign(...(await Promise.all(reqs))));
-  }
-
-  async fetchDetailedRoom(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(roomURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    return {room: camelizeKeys(response.data)};
-  }
-
-  async fetchRoomAttributes(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchRoomAttributesURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    return {roomAttributes: response.data};
-  }
-
-  async fetchRoomAvailability(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchRoomAvailabilityURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    const roomAvailability = camelizeKeys(response.data);
-    roomAvailability.nonbookablePeriods.forEach(period => {
-      period.key = shortid.generate();
-    });
-    roomAvailability.bookableHours.forEach(hours => {
-      hours.key = shortid.generate();
-    });
-
-    return {roomAvailability};
-  }
-
-  async fetchRoomEquipment(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchRoomEquipmentURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    return {roomEquipment: camelizeKeys(response.data)};
-  }
-
-  handleCloseModal = async () => {
-    const {
-      afterCreation,
-      onClose,
-      actions: {fetchEquipmentTypes, fetchRoom, fetchRoomDetails, fetchRoomPermissions},
-    } = this.props;
-    const {wasEverSaved} = this.state;
-    // eslint-disable-next-line react/destructuring-assignment
-    const roomId = this.newRoom ? this.state.newRoomId : this.props.roomId;
-    this.setState({closing: true});
-    if (roomId !== null && wasEverSaved) {
-      await Promise.all([
-        fetchEquipmentTypes(),
-        fetchRoom(roomId),
-        fetchRoomPermissions(roomId),
-        fetchRoomDetails(roomId, true),
-      ]);
-    }
-    onClose(wasEverSaved || afterCreation);
-  };
-
-  handleSubmit = async (data, form) => {
-    let {roomId} = this.props;
+      ].map(x => ({
+        ...x,
+        menuItem: (
+          <Menu.Item key={x.key}>
+            {x.menuItem} {tabsWithError.includes(x.key) ? <Label color="red">Error</Label> : null}
```

Comment:
What about using an icon instead of a text label?

Reply:
Sure, looks good!
![image](https://user-images.githubusercontent.com/12183954/80015694-0a57ae00-84ca-11ea-86ad-1946a62b2276.png)